shopify/utils/utilities.py

```diff
@@ -62,3 +63,14 @@ def __validate_issuer_and_dest_match(cls, decoded_payload):
 
         if issuer_root != dest_root:
             raise MismatchedHostsError()
+
+
+def sanitize_shop_domain(shop_domain, myshopify_domain="myshopify.com"):
+    name = str(shop_domain).lower().strip()
+    if myshopify_domain not in name and "." not in name:
+        name += ".{domain}".format(domain=myshopify_domain)
+    name = re.sub(r"https?://", "", name)
+
+    uri = urlparse("http://{hostname}".format(hostname=name))
```

Comment:
I wonder if we can go with a single RE here:

```suggestion
    matches = re.match(r'(https?://)?({h}\.{d})'.format(h=HOSTNAME_PATTERN, d=myshopify_domain), shop_domain.lower().strip())
    if matches:
        name = matches.group(2)
        return name
```

Reply:
I was playing around with this:
```python
print re.match(r'(https?://)?({h})(\.{d})?'.format(h=r"[a-z0-9][a-z0-9-]*[a-z0-9]", d='myshopify.com'), 'https://test.myshopify.com'.lower().strip()).group(1,2,3)
('https://', 'test', 'myshopify.com')
```

```python
print re.match(r'(https?://)?({h})(\.{d})?'.format(h=r"[a-z0-9][a-z0-9-]*[a-z0-9]", d='myshopify.com'), 'https://test'.lower().strip()).group(1,2,3)
('https://', 'test', None)
```

Seems to work! (Note that I've changed the regex slightly from my initial suggestion)

Reply:
This unfortunately has trouble dealing with all of these bad input cases:

```
"myshop.com",
            "myshopify.com",
            "shopify.com",
            "two words",
            "store.myshopify.com.evil.com",
            "/foo/bar",
            "/foo.myshopify.io.evil.ru",
            "%0a123.myshopify.io ",
            "foo.bar.myshopify.io",
```

Reply:
Hmm yeah the regexes above fail for the following inputs: 
* `shopify.com`: returns `shopifymyshopify.com` (expected `None`)
* `myshop.com`: returns `myshopmyshopify.com` (expected `None`)
* `two words`: returns `twomyshopify.com` (expected `None`)
* `foo.bar.myshopify.io`: returns `foomyshopify.com` (expected `None`)

Reply:
Ah yes, my regex didn't account for cases where the `.` in the domain isn't the start for `.myshopify.com`. That requirement is contextual, so we probably won't be able to handle it with a straight-up regexp. We could rewrite this code around a regex, but I think it would end up looking pretty much the same. LGTM!